amgut/lib/data_access/ag_data_access.py

```diff
@@ -216,186 +200,181 @@ def getAGKitDetails(self, supplied_kit_id):
+                     kit_verification_code, print_results)
+                    SELECT '{0}', kit_id, password, swabs_per_kit,
+                        verification_code, '{1}'
+                        FROM ag_handout_kits WHERE kit_id = %s LIMIT 1
+                    RETURNING ag_kit_id INTO k_id;
+                    FOR bc IN
+                        SELECT barcode
+                        FROM ag_handout_barcodes
+                        WHERE kit_id = %s
+                    LOOP
+                        INSERT  INTO ag_kit_barcodes
+                            (ag_kit_id, barcode, sample_barcode_file)
+                            VALUES (k_id, bc, bc || '.jpg');
+                    END LOOP;
+                    DELETE FROM ag_handout_kits WHERE kit_id = %s;
+                END $do$;
+                """.format(ag_login_id, printresults)
+            TRN.add(sql, [supplied_kit_id] * 3)
+            try:
+                TRN.execute()
+            except psycopg2.IntegrityError:
+                logging.exception('Error on skid %s:' % ag_login_id)
+                return False
+            return True
 
     def get_all_handout_kits(self):
-        conn_handler = SQLConnectionHandler()
-        sql = 'SELECT kit_id FROM ag.ag_handout_kits'
-        return [x[0] for x in conn_handler.execute_fetchall(sql)]
+        with TRN:
+            sql = 'SELECT kit_id FROM ag.ag_handout_kits'
+            TRN.add(sql)
+            return TRN.execute_fetchflatten()
 
     def deleteAGParticipantSurvey(self, ag_login_id, participant_name):
-        # Remove user using old stype DB Schema
-        self.get_cursor().callproc('ag_delete_participant',
-                                   [ag_login_id, participant_name])
-        self.connection.commit()
-
         # Remove user from new schema
-        conn_handler = SQLConnectionHandler()
-        sql = ("SELECT survey_id FROM ag_login_surveys WHERE ag_login_id = "
-               "%s AND participant_name = %s")
-        survey_id = conn_handler.execute_fetchone(
-            sql, (ag_login_id, participant_name))[0]
+        with TRN:
+            sql = """SELECT survey_id
+                     FROM ag_login_surveys
+                     WHERE ag_login_id = %s AND participant_name = %s"""
+            TRN.add(sql, (ag_login_id, participant_name))
+            survey_id = TRN.execute_fetchlast()
 
-        with conn_handler.get_postgres_cursor() as curr:
-            sql = ("DELETE FROM survey_answers WHERE "
-                   "survey_id = %s")
-            curr.execute(sql, [survey_id])
+            sql = "DELETE FROM survey_answers WHERE survey_id = %s"
+            TRN.add(sql, [survey_id])
 
-            sql = ("DELETE FROM survey_answers_other WHERE "
-                   "survey_id = %s")
-            curr.execute(sql, [survey_id])
+            sql = "DELETE FROM survey_answers_other WHERE survey_id = %s"
+            TRN.add(sql, [survey_id])
```

Comment:
Well, one possibility would be something like:

```python
def one_arg(sql, arg):
    with TRN:
        TRN.add(sql, [arg])
```

...or something, which would reduce some code but it does obfuscate some as well